docs/content/v0-upgrade.md

```diff
@@ -0,0 +1,464 @@
+outcome of a rule like `p.a { true }` would change between v0 and v1.0
+without generating an error. Generating errors in this case is preferable to
+changing the semantics of existing rules. Therefore, use of the `if` keyword is
+a requirement in OPA v1.0.
+
+In OPA v1.0, the `if` keyword is only required for rules with a declared body.
+Constants, rules that only consist of a value assignment, do not require `if`.
+The following forms therefore remain valid in OPA v1.0:
+
+| rule       | output in v0           | output in v1.0         |
+| ---------- | ---------------------- | ---------------------- |
+| p := 1     | {“p”: 1}               | {“p”: 1}               |
+| p.a := 1   | {“p”: {“a”: 1}}        | {“p”: {“a”: 1}}        |
+| p.a.b := 1 | {“p”: {“a”: {“b”: 1}}} | {“p”: {“a”: {“b”: 1}}} |
+
+Since the `if` keyword can only be used in front of a rule body, rules with no
+body and no value assignment, i.e. a solitary reference, are not allowed in
+the v1.0 Rego syntax:
+
+| rule  | output in v0              | output in v1.0 |
+| ----- | ------------------------- | -------------- |
+| p     | compile error             | compile error  |
+| p.a   | {“p”: {“a”}}              | compile error  |
+| p.a.b | {“p”: {“a”: {“b”: true}}} | compile error  |
+
+The below table gives examples of v0 valid Rego syntax which are
+invalid in OPA v1.0, along with the equivalent valid syntax in OPA v1.0:
+
+| invalid in v1.0 | v1.0 equivalent            | Note                    |
+| --------------- | -------------------------- | ----------------------- |
+| p { true }      | p if { true }              | Single-value rule       |
+| p.a             | p contains "a"             | Multi-value insertion   |
+| p.a { true }    | p contains "a" if { true } | Multi-value rule        |
+| p.a.b           | p.a.b := true              | Single-value assignment |
+| p.a.b { true }  | p.a.b if { true }          | Single-value rule       |
+
+Following is an example of how to define a rule that generates a set:
+
+```rego
+package play
+
+a contains b if { b := 1 }
+```
+
+When the above rule is evaluated the output is (sets are serialized into arrays
+in JSON):
+
+```json
+{
+  "a": [1]
+}
+```
+
+Following is an example of how to define a rule that generates an object:
+
+```rego
+package play
+
+a[b] if { b := 1}
+```
+
+When the above rule is evaluated the output is:
+
+```json
+{
+  "a": {
+    "1": true
+  }
+}
+```
+
+The requirement of `if` and `contains` keywords remove the ambiguity
+between single-value and multi-value rule declaration. This makes Rego code
+easier to author and read; thereby making it simpler for users to author their
+policies.
+
+### Prohibit duplicate imports
+
+The Rego compiler supports [strict mode](../policy-language/#strict-mode) which
+enforces additional constraints and safety checks during compilation. One of
+these checks is to prohibit duplicate imports where one import shadows another.
```

Comment:
```suggestion
enforces additional constraints and safety checks during compilation. One of
these checks in OPA v0.x was to prohibit duplicate imports where one import shadows another.
```

Or we could flip it: first describe the check, it's implications, and rationale; and end by saying that this was part of strict-mode in v0.x.

Reply:
Reworded this section.